File: python/src/fogbugz_mcp/tools/assign_case.py
from __future__ import annotations

from typing import TYPE_CHECKING

from mcp.server.fastmcp import FastMCP

if TYPE_CHECKING:
    from ..fogbugz_client import FogBugzClient


def register_assign_case(mcp: FastMCP, client: FogBugzClient) -> None:
    @mcp.tool(
        name="assign_case",
        description="Reassign a FogBugz case to a different person.",
        annotations={"destructiveHint": True},
    )
    async def assign_case(case_number: int, assigned_to: str, comment: str | None = None) -> str:
        """Reassign a FogBugz case to a different person."""
        result = await client.assign(case_number, assigned_to, comment)
        case_url = client.get_case_url(result["case"]["ixBug"])
        return f'Case {result["case"]["ixBug"]} assigned to "{assigned_to}".\nLink: {case_url}'

File: src/tools/assign-case.ts
import { z } from "zod";
import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import type { FogBugzClient } from "../fogbugz-client.js";

export function registerAssignCase(
  server: McpServer,
  client: FogBugzClient,
) {
  server.tool(
    "assign_case",
    "Reassign a FogBugz case to a different person.",
    {
      case_number: z.number().describe("The FogBugz case number"),
      assigned_to: z
        .string()
        .describe("Full name of the person to assign the case to"),
      comment: z
        .string()
        .optional()
        .describe("Optional comment to add when reassigning"),
    },
    { destructiveHint: true },
    async ({ case_number, assigned_to, comment }) => {
      const result = await client.assign(case_number, assigned_to, comment);
      const caseUrl = client.getCaseUrl(result.case.ixBug);

      return {
        content: [
          {
            type: "text",
            text: `Case ${result.case.ixBug} assigned to "${assigned_to}".\nLink: ${caseUrl}`,
          },
        ],
      };
    },
  );
}
